frontend/individual_pedigree.tsx

```diff
@@ -0,0 +1,160 @@
+/**
+ * @file This file contains the Pedigree function. This function fetches
+ *       pedigree for a given `id` (parsed from the url).
+ */
+
+import React, { Component, createRef } from 'react'
+import { Link } from "react-router-dom";
+import { get } from './communication';
+import { Network } from 'vis-network';
+import "vis-network/styles/vis-network.css"
+
+
+
+/**
+ * Shows the information of a given individual and the pedigree graph built using the vis-network component
+ */
+export function IndividualPedigree({ id, generations }: { id: string }) {
+  const [pedigree, setPedigree] = React.useState(undefined as any)
+  const [individual, setIndividual] = React.useState(undefined as any)
+  const [generations_input, setGenerations] = React.useState(generations)
+
+  React.useEffect(() => {
+    get(`/api/individual/${id}`).then(
+      data => data && setIndividual(data),
+      error => console.error(error)
+    )
+    get(`/api/pedigree/${id}/${generations_input}`).then(
+      data => data && setPedigree(data),
+      error => console.error(error)
+    )
+  }, [id, generations_input])
+
+  class PedigreeNetwork extends Component {
```

Comment:
This looks with a brief glance like an exact copy of the class from the other file... Perhaps some of the code can be shared?

Reply:
ok

Reply:
solved